test/ess.jl

```diff
@@ -87,4 +126,116 @@
             @test all(ismissing, rhat_array)
         end
     end
+
+    @testset "ESS and R̂ for chains with 2 epochs that have not mixed" begin
+        # checks that splitting yields lower ESS estimates and higher Rhat estimates
+        x = randn(1000, 4, 10) .+ repeat([0, 10]; inner=(500, 1, 1))
+        ess_array, rhat_array = ess_rhat(x; split_chains=1)
+        @test all(x -> isapprox(x, 1; rtol=0.1), rhat_array)
+        ess_array2, rhat_array2 = ess_rhat(x; split_chains=2)
+        @test all(ess_array2 .< ess_array)
+        @test all(>(2), rhat_array2)
+    end
+
+    @testset "ess_rhat(f, x)[1]" begin
+        # we check the ESS estimates by simulating uncorrelated, correlated, and
+        # anticorrelated chains, mapping the draws to a target distribution, computing the
+        # estimand, and estimating the ESS for the chosen estimator, computing the
+        # corresponding MCSE, and checking that the mean estimand is close to the asymptotic
+        # value of the estimand, with a tolerance chosen using the MCSE.
+        ndraws = 1_000
+        nchains = 4
+        nparams = 100
+        x = randn(ndraws, nchains, nparams)
+        mymean(x; kwargs...) = mean(x; kwargs...)
+        @test_throws ArgumentError ess_rhat(mymean, x)
+        estimators = [mean, median, std, mad, Base.Fix2(quantile, 0.25)]
+        dists = [Normal(10, 100), Exponential(10), TDist(7) * 10 - 20]
+        # AR(1) coefficients. 0 is IID, -0.3 is slightly anticorrelated, 0.9 is highly autocorrelated
+        φs = [-0.3, -0.1, 0.1, 0.3, 0.5, 0.7, 0.9]
+        iter = filter(collect(Iterators.product(estimators, dists, φs))) do (f, dist, φ)
+            return !(f === mad) || dist isa Normal
+        end
+        nchecks = nparams * length(iter)
+        α = (0.1 / nchecks) / 2  # multiple correction
+        rng = Random.default_rng()
```

Comment:
What's the purpose of `rng`? It seems you only call `ar1` with the default RNG, so maybe just remove the `rng` argument from `ar1` and sample with `randn(T, ...)` without RNG inside the function?